userspace/libscap/engine/gvisor/proto/pkg/sentry/seccheck/points/common.proto

```diff
@@ -16,6 +16,54 @@ syntax = "proto3";
 
 package gvisor.common;
 
+// Handshake message is used when establishing a connection. Version information
+// is exchanged to determine if the communication can proceed. Each side reports
+// a single version of the protocol that it supports. If they can't support the
+// version reported by the peer, they must close the connection. If the peer
+// version is higher (newer), it should continue to communicate, making the peer
+// responsible to send messages that are compatible with your version. If the
+// peer can't support it, the peer should close the connection.
+//
+// In short:
+//   1. sentry and remote exchange versions
+//   2. sentry continues if remote >= min(sentry)
+//   3. remote continues if sentry >= min(remote)
+//
+// Suppose that peer A is at version 1 and peer B is at 2. Peer A sees that B is
+// at a newer version and continues with communication. Peer B will see that A
+// is at version 1 (older) and will check if it can send messages that are
+// compatible with version 1. If yes, then the communication can continue. If
+// not, A should close the connection.
```

Comment:
In this case, isn't B that should close the connection?

Reply:
Could be a typo in the original gvisor protobuf definition. It seems that B is right though.